examples/moses-pln-synergy/scm/README.md

```diff
@@ -76,16 +76,204 @@ And search for the following
    )
 ...
 
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-neutral-element-hack)
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-dangling-junctor-hack)
-scheme@(guile-user)> (cog-bind pln-rule-equivalence-hack)
-$6 = (SetLink
+scheme@(guile-user)> (cog-bind implication-lambda-distribution-rule)
+...
    (ImplicationLink (stv 1 0.99999982)
-      (PredicateNode "take-treatment-1" (stv 0.1 0.80000001))
-      (PredicateNode "take-compound-A" (stv 0.2 0.80000001))
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (AndLink
+            (EvaluationLink
+               (PredicateNode "take")
+               (ListLink
+                  (VariableNode "$X")
+                  (ConceptNode "treatment-1")
+               )
+            )
+            (EvaluationLink (stv 1 0.99999982)
+               (PredicateNode "contain")
+               (ListLink
+                  (ConceptNode "treatment-1")
+                  (ConceptNode "compound-A")
+               )
+            )
+         )
+      )
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
```

Comment:
One more remark as to why `P(x)->Q(x)` is ill-defined. `P(x)->Q(x)` is essentially in PLN `Prob(Q|P)`. Now how would you define `Prob(Q(x)|P(x))`? A probability is over events (power-sets), but Q(x) and P(x) are not power-sets. OK, let's make a notation abuse and say that `P(x)` is in fact the singleton `{P(x)}`, and let's assume that Omega is discrete so that `{P(x)}` has a well-defined probability. What would be `Prob(Q(x)|P(x))`? It would either 1, if `Q(x)` and `P(x)` denotes the same element in Omega. Or undefined if `Q(x)` and `P(x)` denotes 2 different elements. How useful is that? It isn't... You can wrap a lambda around that, but it's not equal to `Prob(Q(x)|P(x))` (which is what matters to us).

In other words, yes you may try to come up with ways to define `P(x)->Q(x)`, but they are not useful anyway!